cli/src/deploy/mod.rs

```diff
@@ -53,31 +53,31 @@ pub struct AzureCredentials {
 
 #[derive(Deserialize, Debug, Clone, Serialize)]
 #[serde(untagged)]
-pub enum Credentials {
-  GCP(GCPCredentials),
-  AWS(AWSCredentials),
-  Azure(AzureCredentials),
+pub enum CloudCredential {
+  GCP(GCPCredential),
+  AWS(AWSCredential),
+  Azure(AzureCredential),
 }
 
 #[allow(non_snake_case)]
 #[derive(Deserialize, Debug, Serialize)]
-pub struct CredentialsProfile {
-  credentials: Credentials,
+pub struct Credential {
+  credential: CloudCredential,
   cloudProvider: String,
```

Comment:
just realizing that we don't need `cloudProvider` at all - we should probably get rid of the `Credential` struct and have `serde` de/serialize directly into a `CloudCredential` using the `cloudProvider` field to inform which enum variant to construct?

Reply:
I think you are right

Reply:
We don't need it here, but we do need it the deploy service